packages/react-charts/src/components/ChartBar/examples/ChartBar.md

```diff
@@ -163,20 +163,105 @@ import { Chart, ChartBar, ChartVoronoiContainer } from '@patternfly/react-charts
 </div>
 ```
 
+### Basic with right aligned legend (Accessibility Demo)
+
+This demo demonstrate the accessibility function in a Bar chart. You can now focus on the whole chart or the contents in the chart. At the same time, you could use screen reader like VoiceOver to read the chart and contents as you defined in format.
+
+```js
+import React from 'react';
+import { Chart, ChartAxis, ChartBar, ChartGroup, ChartVoronoiContainer } from '@patternfly/react-charts';
+
+function BasicWithRightAlignedLegendAccessibilityDemo() {
+  const data = {
+    cat: [
+      { name: 'Cats', x: '2015', y: 1 },
+      { name: 'Cats', x: '2016', y: 2 },
+      { name: 'Cats', x: '2017', y: 5 },
+      { name: 'Cats', x: '2018', y: 3 }
+    ],
+    dog: [
+      { name: 'Dogs', x: '2015', y: 2 },
+      { name: 'Dogs', x: '2016', y: 1 },
+      { name: 'Dogs', x: '2017', y: 7 },
+      { name: 'Dogs', x: '2018', y: 4 }
+    ],
+    birds: [
+      { name: 'Birds', x: '2015', y: 4 },
+      { name: 'Birds', x: '2016', y: 4 },
+      { name: 'Birds', x: '2017', y: 9 },
+      { name: 'Birds', x: '2018', y: 7 }
+    ],
+    mice: [
+      { name: 'Mice', x: '2015', y: 3 },
+      { name: 'Mice', x: '2016', y: 3 },
+      { name: 'Mice', x: '2017', y: 8 },
+      { name: 'Mice', x: '2018', y: 5 }
+    ]
+  };
```

Comment:
The particular example only has 16 data points. I'd like to see an example with more data points to see how well tabbing works.

The example below has 90 data points.
http://patternfly-react.surge.sh/charts/stack-chart#monthly-data-with-responsive-container

In Cost Management; for example, we have 6 sets of data with a total of 180 data points. However, there are up to 9 charts on a single page. 

My concern is that it may be annoying to tab thru all of that?

Reply:
I also noticed this point and mentioned it in the additional issues. Still need more suggestions about how we could implement this!

Reply:
I think this is a really good point - I think it would be tedious to tab through 90-250k data points. I've been trying to do some research on how other accessible charts do this, and it doesn't seem like there's a firm standard. However, my favorite chart I've seen so far in terms of how they handle this situation is this [Highcharts demo](https://www.highcharts.com/demo/bar-basic). 

What I like about their implementation is that they have two axes/directions for navigation, columns and rows. Left/right arrow keys navigate bars within a group (and only the one group), and up/down arrow keys navigate the same category from group to group, then starts over with the next category. The tab key itself navigates to the main regions (like starting data point then legend then to the next element). This would account for a case where you have a ton of data points because you only tab a few times per chart but can use arrow keys if you want to dig into the data points deeper. I'm guessing we could use event handlers to manage this focus similar to how we've handled [dropdown](https://github.com/patternfly/patternfly-react/blob/21a343deedf3c30f6dc63f34b23f10cf93103611/packages/react-core/src/components/Dropdown/InternalDropdownItem.tsx#L113) keyboard interaction, [select](https://github.com/patternfly/patternfly-react/blob/6abd540e61ad0226b359e2f91f3be6042dd76f2a/packages/react-core/src/components/Select/SelectOption.tsx#L104), etc.

Reply:
Using the up/down arrow keys would certainly be better than tabbing. Although, keep in mind that the Highcharts demo is also a simple bar chart with limited (~12) data points.

I'd also like to explore the use of a hidden HTML table. The idea is that screen readers could choose to navigate thru all the data if desired, but in a more standard way, via a traditional HTML table. This would free typical users from having all those data points added to the tab index.

Reply:
Thanks @jessiehuff @dlabrecq , I will also do some research on both Highcharts and hidden HTML table.